Firestore/core/src/firebase/firestore/core/firestore_client.cc

```diff
@@ -377,11 +384,12 @@ void FirestoreClient::GetDocumentsFromLocalCache(
   auto shared_callback = absl::ShareUniquePtr(std::move(callback));
   auto shared_this = shared_from_this();
   worker_queue()->Enqueue([shared_this, query, shared_callback] {
-    DocumentMap docs = shared_this->local_store_->ExecuteQuery(query.query());
+    QueryResult query_result = shared_this->local_store_->ExecuteQuery(
+        query.query(), /* use_previous_results= */ true);
 
-    View view(query.query(), DocumentKeySet{});
+    View view(query.query(), std::move(query_result.remote_keys()));
```

Comment:
I tried to make this work but I could not figure out how to move both `remote_keys` and `documents` (since the most straightforward way requires calling std::move() twice). It seems like the ref-qualifier is only used if I invoke std::move() and I don't assign it to an lvalue (which would contain the result of a single std::move()). Once assigned to an lvalue, `query_data.remote_keys()` or `query_data.documents()` seem to use the const reference and create a copy.

So I could make this work by only calling std::move() for one of these getters, but then I have to flip a coin and decide whether I want to move `remote_keys` or `documents`. Instead. I opted to copy both for now.